src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,73 @@
+/* @flow */
+import * as React from 'react';
+
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import {
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import type { ResultShelfType } from 'amo/reducers/home';
+
+import './styles.scss';
+
+type Props = {|
+  addonInstallSource?: string,
+  loading: boolean,
+  placeholderCount: number,
+  shelves: Array<ResultShelfType>,
+|};
+
+export default class HomepageShelves extends React.Component<Props> {
+  static defaultProps = {
+    placeholderCount: LANDING_PAGE_EXTENSION_COUNT,
+  };
+
+  render() {
+    const {
+      addonInstallSource,
+      loading,
+      placeholderCount,
+      shelves,
+    } = this.props;
+
+    const homepageShelves = shelves.map((shelf) => {
+      const { addons, endpoint, footer_text, title } = shelf;
+      const header = title.replace(/\s/g, '');
```

Comment:
After fixing the problem with the `versions` reducer, I noticed this is now causing an error, because your code is assuming `title` will be a string, but it's actually now a localized string (i.e., an object) returned from the API. You'll need to deal with these now localized fields in the routine where you create the internal representation of the shelves, and that should fix this line without requiring any changes to this file.